pkg/controller/dnsendpoint/common.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dnsendpoint
+
+import (
+	"net"
+	"sort"
+
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	feddnsv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/multiclusterdns/v1alpha1"
+)
+
+const (
+	// minDNSTTL is the minimum safe DNS TTL value to use (in seconds).  We use this as the TTL for all DNS records.
```

Comment:
I understand this was previously defined in `pkg/controller/servicednsendpoint/servicednsendpoint_controller.go`, and should in no way affect the mergability of this PR; however, as I'm trying to come up to speed, I'm trying to determine if this comment this accurate. From what I can tell, this would more accurately be named `defaultDNSTTL` as I am not able to find a comparison that would prevent TTLs from being defined less than this number. It appears that DNS records are being forcefully set to this value. Additionally, if `180` is indeed the minimum DNS TTL, I am curious the reason this was chosen.

Reply:
you are right, the constant should be renamed `defaultDNSTTL`. User can override the default TTL value for DNS records in API types. The code is updated now. Thanks for pointing this out.